packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -295,6 +321,28 @@ abstract class ImageStreamCompleter extends Diagnosticable {
   ///
   /// If no error listeners are attached, a [FlutterError] will be reported
   /// instead.
+  ///
+  /// The `context` should be a string describing where the error was caught, in
+  /// a form that will make sense in English when following the word "thrown",
+  /// as in "thrown while obtaining the image from the network" (for the context
+  /// "while obtaining the image from the network").
+  ///
+  /// The `exception` is the error being reported; the `stack` is the
+  /// [StackTrace] associated with the exception.
+  ///
+  /// The `informationCollector` is a callback (of type [InformationCollector])
+  /// that is called when the exception is used by [FlutterError.reportError].
+  /// It is used to obtain further details to include in the logs, which may be
+  /// expensive to collect, and thus should only be collected if the error is to
+  /// be logged in the first place.
+  ///
+  /// The `silent` argument should be set to true if the error is one that is
+  /// expected to be encountered in release builds, for example network errors.
+  /// It causes the exception to not be reported to the logs in release builds,
```

Comment:
I'd flip the 2 sentences. Describe what it does first before suggesting how it should be used.